src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -351,56 +370,56 @@ void func_80A7C86C(EnInsect* this, GlobalContext* globalCtx) {
     this->skelAnime.playSpeed = CLAMP(this->actor.speedXZ * 1.6f, 0.8f, 1.9f);
     SkelAnime_Update(&this->skelAnime);
 
-    if (this->unk_31A <= 0 || !sp38) {
-        func_80A7C3A0(this);
-    } else if ((this->unk_314 & 1) && (this->actor.bgCheckFlags & 0x40)) {
-        func_80A7CE60(this);
+    if (this->actionTimer <= 0 || !playerIsClose) {
+        EnInsect_SetupCrawl(this);
+    } else if ((this->flags & INSECT_FLAG_TEMP_AND_ALIVE) && (this->actor.bgCheckFlags & 0x40)) {
+        EnInsect_SetupSink(this);
     }
 }
 
-void func_80A7CA64(EnInsect* this) {
-    this->unk_31A = 200;
+void EnInsect_SetupCaught(EnInsect* this) {
+    this->actionTimer = 200;
 
     Actor_SetScale(&this->actor, 0.001f);
 
     this->actor.draw = NULL;
     this->actor.speedXZ = 0.0f;
 
-    func_80A7BF58(this);
+    EnInsect_SetupAnimation(this);
 
     this->skelAnime.playSpeed = 0.3f;
-    this->actionFunc = func_80A7CAD0;
-    this->unk_314 &= ~0x100;
+    this->actionFunc = EnInsect_UpdateCaught;
```

Comment:
```suggestion
    this->actionFunc = EnInsect_Caught;
```
same with `UpdateDropped`, can drop the `Update` part